core/src/main/scala/kafka/server/KafkaApis.scala

```diff
@@ -425,35 +426,59 @@ class KafkaApis(val requestChannel: RequestChannel,
       requestHelper.sendMaybeThrottle(request, offsetCommitRequest.getErrorResponse(Errors.UNSUPPORTED_VERSION.exception))
       CompletableFuture.completedFuture[Unit](())
     } else {
+      val topicResolver = TopicResolver.fromTopicIds(metadataCache.topicNamesToIds())
```

Comment:
Move the `TopicResolver` in the `MetadataCache` or create it without copying the map of topic ids as this is costly.